packages/db/postgres/functions.sql

```diff
@@ -250,6 +250,95 @@ BEGIN
 END
 $$;
 
+DO 
+$$
+BEGIN
+  IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'used_storage') THEN
+    CREATE TYPE used_storage AS (uploaded TEXT, pinned TEXT, total TEXT);
+  END IF;
+END
+$$;
+
+CREATE OR REPLACE FUNCTION user_used_storage(query_user_id BIGINT) 
+RETURNS used_storage
+LANGUAGE plpgsql
+AS
+$$
+DECLARE 
+  used_storage  used_storage;
+  uploaded      BIGINT;
+  pinned        BIGINT;
+  total         BIGINT;
+BEGIN
+  uploaded := 
+    (
+      SELECT COALESCE(SUM(c.dag_size), 0)
+      FROM upload u
+      JOIN content c ON c.cid = u.content_cid
+      WHERE u.user_id = query_user_id::BIGINT
+      AND u.deleted_at is null
```

Comment:
Probably for a separate issue/PR, but here we're counting also "Uploads" that aren't and possibly will never be actually pinned.
We should take the same approach we're taking for "pinned".
In fact, the same query should be re-used for both.

Reply:
Sounds like a good candidate for another ticket definitely.